spec/requests/api/v1/customers_spec.rb

```diff
@@ -0,0 +1,341 @@
+      produces "application/json"
+
+      response "200", "Customers list" do
+        param(:enterprise_id) { enterprise1.id }
+        schema "$ref": "#/components/schemas/resources/customers_collection"
+
+        run_test!
+      end
+    end
+
+    describe "returning results based on permissions" do
+      context "as guest user" do
+        before { login_as nil }
+
+        it "returns no customers" do
+          get "/api/v1/customers"
+          expect(json_response_ids).to eq []
+        end
+
+        it "returns not even customers without user id" do
+          customer3.update!(user_id: nil)
+
+          get "/api/v1/customers"
+          expect(json_response_ids).to eq []
+        end
+      end
+
+      context "as an enterprise owner" do
+        before { login_as enterprise1.owner }
+
+        it "returns customers of enterprises the user manages" do
+          get "/api/v1/customers"
+          expect(json_response_ids).to eq [customer1.id.to_s, customer2.id.to_s]
+        end
+      end
+
+      context "as another enterprise owner" do
+        before { login_as enterprise2.owner }
+
+        it "returns customers of enterprises the user manages" do
+          get "/api/v1/customers"
+          expect(json_response_ids).to eq [customer3.id.to_s]
+        end
+      end
+
+      context "with ransack params searching for specific customers" do
+        before { login_as enterprise2.owner }
+
+        it "does not show results the user doesn't have permissions to view" do
+          get "/api/v1/customers", params: { q: { id_eq: customer2.id } }
+
+          expect(json_response_ids).to eq []
+        end
+      end
+    end
+
+    describe "pagination" do
+      it "renders the first page" do
+        get "/api/v1/customers", params: { page: "1" }
+        expect(json_response_ids).to eq [customer1.id.to_s, customer2.id.to_s]
+      end
+
+      it "renders the second page" do
+        get "/api/v1/customers", params: { page: "2", per_page: "1" }
+        expect(json_response_ids).to eq [customer2.id.to_s]
+      end
+
+      it "renders beyond the available pages" do
+        get "/api/v1/customers", params: { page: "2" }
+        expect(json_response_ids).to eq []
+      end
+
+      it "informs about invalid pages" do
+        get "/api/v1/customers", params: { page: "0" }
+        expect(json_response_ids).to eq nil
+        expect(json_error_detail).to eq 'expected :page >= 1; got "0"'
+      end
+    end
+
+    post "Create customer" do
+      tags "Customers"
+      consumes "application/json"
+      produces "application/json"
+
+      parameter name: :customer, in: :body, schema: {
+        type: :object,
+        properties: CustomerSchema.writable_attributes,
+        required: CustomerSchema.required_attributes
+      }
+
+      response "201", "Minimal customer created" do
+        param(:customer) do
+          {
+            email: "test@example.com",
+            enterprise_id: enterprise1.id.to_s
+          }
```

Comment:
I think that we are supposed to send parameters like this:
```suggestion
          {
            data: {
              type: "customer",
              attributes: {
                email: "test@example.com",
                enterprise_id: enterprise1.id.to_s
              }
            }
          }
```

> A resource can be created by sending a POST request to a URL that represents a collection of resources. The request MUST include a single [resource object](https://jsonapi.org/format/#document-resource-objects) as primary data. The [resource object](https://jsonapi.org/format/#document-resource-objects) MUST contain at least a type member.

https://jsonapi.org/format/#crud-creating